syntax = "proto3";

package fraud_detection;

service FraudDetectionService {
    rpc CheckFraud (FraudRequest) returns (FraudResponse);
}

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}

message OrderItem {
    string name = 1;
    int32 quantity = 2;
}

message BillingAddress {
    string street = 1;
    string city = 2;
    string state = 3;
    string zip = 4;
    string country = 5;
}

message FraudRequest {
    User user = 1;
    CreditCard credit_card = 2;
    string user_comment = 3;
    repeated OrderItem items = 4;
    BillingAddress billing_address = 5;
    string shipping_method = 6;
    bool gift_wrapping = 7;
    bool terms_accepted = 8;
}

message FraudResponse {
    bool is_fraud = 1;
    optional string error_message = 2;
}
